root/static/scripts/tests/Control/URLCleanup.js

```diff
@@ -5279,6 +5287,46 @@ testData.forEach(function (subtest, i) {
         tested = true;
       }
     }
+    if (subtest.expected_error) {
+      const relationshipType = subtest.input_relationship_type ||
+        subtest.expected_relationship_type;
+      if (!relationshipType) {
+        st.fail(
+          'Test is invalid: "expected_error" is specified with neither "expected_relationship_type" nor "input_relationship_type".',
+        );
+        st.end();
+        return;
+      }
+      if (!subtest.input_entity_type) {
+        st.fail(
+          'Test is invalid: "expected error" is specified without "input_entity_type".',
+        );
+        st.end();
+        return;
+      }
+      const cleanUrl = subtest.expected_clean_url || actualCleanUrl;
+      const checker = new Checker(cleanUrl, subtest.input_entity_type);
+      const validationResult = checker.checkRelationship(
+        LINK_TYPES[relationshipType],
+        subtest.input_entity_type,
+      );
+      if (subtest.expected_error.error === undefined) {
+        st.ok(
+          validationResult.error === undefined,
+          'Default error message will be used as expected',
+        );
+      } else {
+        st.ok(
+          validationResult.error.includes(subtest.expected_error.error),
+          'Error message contains expected string',
+        );
+      }
+      st.equal(
+        validationResult.target,
+        subtest.expected_error.target,
+        'Error target matches expected target',
+      );
```

Comment:
I suggest to move this part of the code to a separate function to improve readability.
(The same should have been done with validation tests really.)
Can be done after merging this pull request if in a hurry.

Reply:
I made the change in a second commit, but do let me know if that looks like what you expected.